### PR TITLE
Correct some typos and do some small adjustments

### PR DESCRIPTION
**Pull request by cd1989**
Signed-off-by: cd1989 <chende@caicloud.io>

### CODE REVIEW

File: src/replication/ng/operation/flow/stage.go [hunk 1]
```diff
@@ -59,8 +59,8 @@ func initialize(policy *model.Policy) (adp.Adapter, adp.Adapter, error) {
 
 // fetch resources from the source registry
 func fetchResources(adapter adp.Adapter, policy *model.Policy) ([]*model.Resource, error) {
-	resTypes := []model.ResourceType{}
-	filters := []*model.Filter{}
+	var resTypes []model.ResourceType
+	var filters []*model.Filter
```
Comment (cd1989) [inline src/replication/ng/operation/flow/stage.go]: Use nil slices over empty slices: https://programming.guide/go/nil-slice-vs-empty-slice.html

File: src/replication/ng/operation/flow/stage.go [hunk 2]
```diff
@@ -185,28 +186,28 @@ func filterResources(resources []*model.Resource, filters []*model.Filter) ([]*m
...
-		namespace, err := srcAdapter.GetNamespace(srcResource.Metadata.Namespace)
-		if err != nil {
-			return nil, err
-		}
-		namespaces = append(namespaces, namespace)
-	}
-
 	if len(dstNamespace) != 0 {
-		namespaces = []*model.Namespace{
+		return []*model.Namespace{
```
Comment (ywk253100) [inline src/replication/ng/operation/flow/stage.go]: We have to set the metadata of destination namespace according to the metadata of source namespaces, so we must get the source namespaces first. And I have add a TODO in line 195.

File: src/replication/ng/operation/flow/stage.go [hunk 3]
```diff
@@ -185,28 +186,28 @@ func filterResources(resources []*model.Resource, filters []*model.Filter) ([]*m
...
+	for _, srcResource := range srcResources {
+		namespace, err := srcAdapter.GetNamespace(srcResource.Metadata.Namespace)
+		if err != nil {
+			return nil, err
 		}
+		namespaces = append(namespaces, namespace)
 	}
 
-	log.Debug("assemble the destination namespaces completed")
+	log.Debugf("assemble the destination namespaces completed: %v", namespaces)
```
Comment (ywk253100) [inline src/replication/ng/operation/flow/stage.go]: It will print just the address in memory rather than the names

Comment (cd1989) [reply src/replication/ng/operation/flow/stage.go]: You are right

**Pull request merged**